articles/model-anatomy.adoc

```diff
@@ -0,0 +1,644 @@
+class b as "2:House"
+
+class a as "33:Address" {
+  street: String = "Main Street",
+  zipCode: String = "12345"
+}
+
+class c1 as "23:Room" {
+  title: String = "Hallway"
+}
+
+class c2 as "1:Bedroom" {
+  title: String = "Master bedroom"
+}
+
+class c3 as "122:Bedroom" {
+  title: String = "Guest room"
+}
+
+b *- a: address
+
+b *-- c1: rooms [0]
+b *-- c2: rooms [1]
+b *-- c3: rooms [2]
+
+----
+
+A containment's type refers to a classifier, and describes which nodes can be part of that containment.
+Only nodes that have a compatible classifier can be part of that containment.
+
+On serialization level, we store the contained node's id for a containment.
+A containment is invalid if the contained node's id cannot be resolved.
+
+[[references]]
+=== References
+
+A node can point to other nodes it has some relation to.
+For example, a `Project` might refer to its `mainResponsible: Employee`, or to all its `externalParties: BusinessPartner`.
+The node containing the reference is called _source_, the referred node _target_ of the reference.
+A node can be referenced by none, one, or many other nodes.
+It can be referenced by itself.
+A node can reference another node (or itself) more than once.
+
+[plantuml, project, svg]
+----
+hide empty members
+
+class pr as "6:Project" {
+  description: String = "Settle beyond this galaxy"
+}
+
+class coop as "9:TemporaryEmployee" {
+  name: String = "Cooper"
+}
+
+class esa as "87:BusinessPartner" {
+  title: String = "ESA"
+  budget: Integer = 23
+}
+
+class jaxa as "40:BusinessPartner" {
+  title: String = "JAXA"
+  budget: Integer = 42
+}
+
+pr ..> esa: externalParties [0]
+pr ..> jaxa: externalParties [1]
+
+pr .> coop: mainResponsible
+
+class pr2 as "90:Project" {
+  description: String = "Grow corn"
+}
+
+coop <. pr2: mainResponsible
+----
+
+A node's classifier describes the valid references.
+This description includes the references’ _name_, _key_, whether it is _optional_, whether we allow _multiple_ targets in the reference, and its _type_.
+All of them mean the same as for <<containments, containments>>, respectively.
+For multiple references, we keep track which target is listed at which position.
+The same target can appear several times in a multiple reference.
+This can help, for example, to describe steps: `steps: Task[1..*] = {generate, saveAll, compile, saveAll}` refers to Task `saveAll` twice.
+
+LionWeb does not support bi-directional references.
+They are very hard to maintain, and also hard to process (i.e. if we want to traverse the whole model).
+
+==== ResolveInfo
+For references, we store the target node's id, and additionally a `resolveInfo`.
+ResolveInfo is a string that somehow describes the target node.
+Typical candidate would be the target's node `name` (if the target node has such a property).
+To support this typical default behavior, LionWeb has a built-in interface `INamed` with one property `name: String` that might be implemented by any classifier.
```

Comment:
This sentence comes a bit out of the blue: there has been no introduction of interfaces and of built-in classifiers so far. Btw this built-in interface `INamed` is something that was quite hard to match in Rascal (while even the built-in `Node` has the corresponding type in Rascal).

Reply:
Is INamed actually introduced for resolving references? I thought it has more generic use. 
Yes, a footnote will be better.

Reply:
It's not introduced, and yes, it has more generic use -- for everything that has a name. I think we propose to use it as default behavior for ResoveInfo in some doc.